kubeconfig.go

```diff
@@ -0,0 +1,93 @@
+package kubeconfig
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Config represents the configuration used to create a new kubeconfig library instance.
+type Config struct {
+	RestConfig rest.Config
+	Logger     micrologger.Logger
+	K8sClient  kubernetes.Interface
+}
+
+// TenantCluster provides functionality for connecting to tenant clusters.
+type KubeConfig struct {
+	logger    micrologger.Logger
+	k8sClient kubernetes.Interface
+}
+
+// New creates a new tenant cluster service.
+func New(config Config) (*KubeConfig, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+
+	g := &KubeConfig{
+		logger:    config.Logger,
+		k8sClient: config.K8sClient,
+	}
+
+	return g, nil
+}
+
+// NewG8sClientFromSecret returns a generated clientset based on the kubeconfig stored in a secret.
+func (k KubeConfig) NewG8sClientFromSecret(ctx context.Context, secretName, secretNamespace string) (versioned.Interface, error) {
+	restConfig, err := k.getRESTConfigFromSecret(ctx, secretName, secretNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := versioned.NewForConfig(restConfig)
+	if err != nil {
+		return nil, microerror.Maskf(err, "rest.RESTClientFor")
```

Comment:
`rest.RESTClientFor` does not match the called function `versioned.NewForConfig`